[package] Add better debugging for torch.package

PaliC opened the pull request:
Summary:
Makes torch.package debugging more transparent by
1. Pointing out not implictily externed modules in the standard library.
2. Creating a debug mode for users to find the source of broken modules.

Test Plan: Run package tests

Differential Revision: D42728753

kurman commented on torch/package/package_exporter.py:
```diff
@@ -154,7 +154,25 @@ def __init__(self, dependency_graph: DiGraph):
+                if debug:
+                    module_path = dependency_graph.first_path(module_name)
+                    message.write(f"      A path to {module_name}: {' -> '.join(module_path)}")
+        if not debug:
```
shouldn't this be in the for loop?

PaliC replied:
This error message specifically isn't related to an individual module and is just advertising the debug functionality. It wouldn't make sense to stick it in the loop.

PaliC commented:
@pytorchbot merge -g

PaliC committed:
[package] Add better debugging for torch.package (#92939)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/92939

Makes torch.package debugging more transparent by
1. Pointing out not implictily externed modules in the standard library.
2. Creating a debug mode for users to find the source of broken modules.

Test Plan: Run package tests

Reviewed By: kurman

Differential Revision: D42728753

fbshipit-source-id: 78b18d330dc71f5aef8c6995d4479e6bddcee7e1

PaliC commented:
@pytorchbot merge -g

PaliC commented:
@pytorchbot merge -g
